pkgs/os-specific/linux/nixos-rebuild/nixos-rebuild.sh

```diff
@@ -8,6 +8,24 @@ shopt -s inherit_errexit
 
 export PATH=@path@:$PATH
 
+
+# open a new file descriptor called 3
+# save all '1>&3 2>&3' or piped to msgPipe or 'msg message' to systemd journal
+# 'journalctl -t nixos-rebuild' to inspect
+# tee is used so we still get output to the terminal
+# check if systemd-cat exists and if systemctl exits with 0. if it does not
+# then systemd isn't running(chroot) and we cannot use systemd-cat.
+command -v systemd-cat &>/dev/null && systemctl &>/dev/null && exec 3> >(tee >(systemd-cat -t nixos-rebuild)) || exec 3>&1
+
+msg() {
+  echo -e "$@" 1>&3 2>&3
+}
+
+msgPipe() {
+  read -r in
```

Comment:
This only reads the first line of stdin. If you want to handle multiple lines and/or stderr you'll need a few more lines.